adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -544,6 +571,34 @@ public AuthenticationResult refreshToken(String refreshToken) throws IOException
         return postMessage(requestMessage, headers);
     }
 
+    public AuthenticationResult refreshTokenUsingAssertion(String samlAssertion, String assertionType)
```

Comment:
Can we add nullability (`@Nullable` and/or `@NonNull`) and `final` on the parameters to this method?

Nullability annotations should be applied based on whether you expect/support providing `null` as params to this function

Reply:
> Can we add nullability (`@Nullable` and/or `@NonNull`) and `final` on the parameters to this method?

Yes both the parameters are expected to not be null. Will add that